training/src/app/modules/chuck-norris/services/joke/joke.service.ts

```diff
@@ -0,0 +1,41 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from "@angular/common/http";
+import { catchError, Observable, of, throwError } from "rxjs";
+import { Joke } from "../../models/joke.modle";
+
+@Injectable({
+	providedIn: 'root'
+})
+export class JokeService {
+	private RandomJokeUrl = 'https://api.chucknorris.io/jokes/random';
+	private jokeListUrl   = 'https://api.chucknorris.io/jokes/search'
+	constructor(private http: HttpClient) {
+	}
+
+	public getRandomJoke(): Observable<Joke> {
+		return this.http.get<Joke>(this.RandomJokeUrl).pipe(
+			catchError(JokeService.handleError)
+		);
+	}
+
+	public getSearchListJokes(searchKey: string): Observable<any> {
+		if (searchKey.length > 2 && searchKey.length < 120) {
```

Comment:
this is not the place to check it.
we should add validation on the input itself, and prevent calling the service if not valid